### PR TITLE
Feature/cmn 293 digital list not stackable

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

https://platform.dandigbib.org/issues/5389

#### Description

Now items in the "group modal" opens another modal
And digital materials are no longer stacked (yay)

- Update wiremock data
- Added listheader component
- Added ToggleListViewButtons component, which are the buttons in the header toggling view (stack/list)
- Deletede paginatedloans, instead using list 
- moved selectedablematerial (just has its own folder now)
- added status message, to make selectedablematerial easier to readable
- renamed stacked -> stack

#### Screenshot of the result

image

#### Checklist

- [ ] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [ ] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [ ] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

If you have any further comments or questions for the reviewer them please add them here.

### CODE REVIEW

File: src/apps/loan-list/list/loan-list-items.tsx [hunk 1]
```diff
@@ -41,10 +32,11 @@ const LoanListItems: FC<LoanListItemProps> = ({
...
         {t("loanListDueDateModalAriaLabelText")}
       </div>
-      {view === "stacked" &&
-        localDueDates.map((uniqueDueDate: string | null) => {
+      {view === "stack" &&
+        dueDates &&
+        dueDates.map((uniqueDueDate: string) => {
           // Stack items:
-          // if multiple items have the same due date, they are "stacked"
+          // if multiple items have the same due date, they are "stack"
```
Comment (turegjorup) [inline src/apps/loan-list/list/loan-list-items.tsx:39]: they are a "stack"

File: src/apps/loan-list/list/ToggleListViewButtons.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  }, [setView]);
+
+  return (
+    <div className="dpl-list-buttons__buttons">
+      {/* <div
+        id="renew-multiple-modal"
+        className="dpl-list-buttons__screen-reader-description"
+      >
+        {t("loanListRenewMultipleButtonExplanationText")}
+      </div> */}
```
Comment (turegjorup) [inline src/apps/loan-list/list/ToggleListViewButtons.tsx:60]: Either remove, add `@todo` or uncomment

**Review by turegjorup — CHANGES_REQUESTED** [review]
Minor changes required

File: src/apps/loan-list/list/ToggleListViewButtons.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  }, [allLoansId, open]);
+
+  useEffect(() => {
+    const modalString = getUrlQueryParam("modal");
+
+    // If the queryparme is a date, the view should be stacked
+    // because the modal it opens can only be opened in the stacked view
+    if (isDate(modalString)) {
+      setView("stack");
+    }
```
Comment (spaceo) [inline src/apps/loan-list/list/ToggleListViewButtons.tsx:50]: This seems really funky. I don't have an overview of the functionality but I think there is a code smell here.
Maybe you disagree but this seems to be fragile and could break if it gets extended.
Could it be an alternative to set an url parameter called view?

File: src/apps/loan-list/list/ToggleListViewButtons.tsx [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        id="renew-multiple-modal"
+        className="dpl-list-buttons__screen-reader-description"
+      >
+        {t("loanListRenewMultipleButtonExplanationText")}
+      </div>
+      <div className="dpl-list-buttons__buttons__button">
+        <button
+          onClick={() => setViewHandler("list")}
+          className={`dpl-icon-button ${
+            view === "list" ? "dpl-icon-button--selected" : ""
```
Comment (spaceo) [inline src/apps/loan-list/list/ToggleListViewButtons.tsx:65]: pleas use clsx here

File: src/apps/loan-list/list/ToggleListViewButtons.tsx [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+          type="button"
+          aria-label={t("loanListAriaLabelListButtonText")}
+        >
+          <IconList />
+        </button>
+      </div>
+      <div className="dpl-list-buttons__buttons__button">
+        <button
+          className={`dpl-icon-button ${
+            view === "stack" ? "dpl-icon-button--selected" : ""
```
Comment (spaceo) [inline src/apps/loan-list/list/ToggleListViewButtons.tsx:77]: pleas use clsx here

File: src/apps/loan-list/list/ToggleListViewButtons.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+          }`}
+          id="test-stack"
```
Comment (spaceo) [inline src/apps/loan-list/list/ToggleListViewButtons.tsx:79]: If this is for Cypress, please use data-cy

File: src/apps/loan-list/list/loan-list-items.tsx [hunk 7]
```diff
@@ -41,10 +32,11 @@ const LoanListItems: FC<LoanListItemProps> = ({
       >
         {t("loanListDueDateModalAriaLabelText")}
       </div>
-      {view === "stacked" &&
-        localDueDates.map((uniqueDueDate: string | null) => {
+      {view === "stack" &&
+        dueDates &&
+        dueDates.map((uniqueDueDate: string) => {
```
Comment (spaceo) [inline src/apps/loan-list/list/loan-list-items.tsx:37]: not fund of those untyped date formats, but we have talked about that ;) 
It's a future todo

File: src/apps/loan-list/list/loan-list.tsx [hunk 8]
```diff
@@ -82,39 +80,56 @@ const LoanList: FC<LoanListProps> = ({ pageSize }) => {
...
       )}
-      {physicalLoans.length === 0 && digitalLoans.length === 0 && (
-        <EmptyList
-          emptyListText={t("loanListDigitalPhysicalLoansEmptyListText")}
-        />
-      )}
+      {Array.isArray(physicalLoans) &&
+        physicalLoans.length === 0 &&
+        Array.isArray(digitalLoans) &&
+        digitalLoans.length === 0 && (
```
Comment (spaceo) [inline src/apps/loan-list/list/loan-list.tsx:128]: You could create a helper function here or I think lodash have an utility function for it too:
```typescript
const loansAreEmpty = (loans: something[]) => Array.isArray(loans) && !loans.length
```

File: src/apps/loan-list/materials/selectable-material/selectable-material.tsx [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+}) => {
+  const t = useText();
+  const { open } = useModalButtonHandler();
+  const { dueDate, faust, identifier } = loan;
+  const { authors, materialType, year, title } = material || {};
+
+  const selectListMaterial = useCallback(
+    (e: MouseEvent) => {
+      e.stopPropagation();
+      open(faust || identifier || "");
```
Comment (spaceo) [inline src/apps/loan-list/materials/selectable-material/selectable-material.tsx:37]: How can you call open on an empty string?

File: src/apps/loan-list/materials/selectable-material/selectable-material.tsx [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+          className={`list-materials ${
+            disabled ? "list-materials--disabled" : ""
+          }`}
+        >
+          <div className="mr-32">
+            {faust && onChecked && (
+              <CheckBox
+                onChecked={() => onChecked(faust)}
+                id={faust}
+                selected={materialsToRenew?.indexOf(faust) > -1}
```
Comment (spaceo) [inline src/apps/loan-list/materials/selectable-material/selectable-material.tsx:55]: What happens here if materialsToRenew is undefined? indexOf will be called on undefined?

File: src/apps/loan-list/materials/utils/status-badge.tsx [hunk 11]
```diff
@@ -3,7 +3,7 @@ import statusThreshold from "../../../../core/configuration/status-thresholds.js
 import { daysBetweenTodayAndDate } from "../../../../core/utils/helpers/general";
 
 interface StatusBadgeProps {
-  dueDate: string;
+  dueDate: string | null | undefined;
```
Comment (spaceo) [inline src/apps/loan-list/materials/utils/status-badge.tsx:6]: Is dueDate required here?
If not you could write the same like:
```typescript
dueDate?: string | null;

File: src/apps/loan-list/modal/material-details.tsx [hunk 12]
```diff
@@ -54,7 +59,21 @@ const MaterialDetails: FC<MaterialDetailsProps & MaterialProps> = ({
...
       </ModalDetailsHeader>
-      {faust && <RenewButton faust={faust} renewable={isRenewable} />}
+      {!isDigital(loan) && faust && (
+        <RenewButton faust={faust} renewable={isRenewable} />
+      )}
+      {isDigital(loan) && (
+        <div className="modal-details__buttons">
+          {/* todo create a component for ereolen-redirect (also replace url) */}
+          <Link
+            href={new URL("https://ereolen.dk/user/me/")}
```
Comment (spaceo) [inline src/apps/loan-list/modal/material-details.tsx:69]: Should we make this a data prop url so it can be configured by the cms?

**Review by spaceo — CHANGES_REQUESTED** [review]
Looks good.
Meanwhile... Have some comments 🧛

File: src/apps/loan-list/modal/material-details.tsx (hunk 12 above)
Comment (sinejespersen) [reply src/apps/loan-list/modal/material-details.tsx:69]: {/* todo create a component for ereolen-redirect (also replace url) */} <-

File: src/apps/loan-list/materials/utils/status-badge.tsx (hunk 11 above)
Comment (sinejespersen) [reply src/apps/loan-list/materials/utils/status-badge.tsx:6]: yesss

File: src/apps/loan-list/materials/selectable-material/selectable-material.tsx (hunk 10 above)
Comment (sinejespersen) [reply src/apps/loan-list/materials/selectable-material/selectable-material.tsx:55]: It returns undefined, should i wrap it in a Boolean(...

File: src/apps/loan-list/materials/selectable-material/selectable-material.tsx (hunk 9 above)
Comment (sinejespersen) [reply src/apps/loan-list/materials/selectable-material/selectable-material.tsx:37]: would you prefer if(faust||identifier) { open(...)}

File: src/apps/loan-list/list/ToggleListViewButtons.tsx (hunk 6 above)
Comment (sinejespersen) [reply src/apps/loan-list/list/ToggleListViewButtons.tsx:79]: Done

File: src/apps/loan-list/list/ToggleListViewButtons.tsx (hunk 3 above)
Comment (sinejespersen) [reply src/apps/loan-list/list/ToggleListViewButtons.tsx]: removed

**Review by spaceo — APPROVED** [review]

**Review by turegjorup — APPROVED** [review]

**Pull request merged**